<?xml version="1.0"?>
<launch>
	    <!-- 设置launch文件的参数 -->
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>

	<!--运行gazebo仿真环境-->
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
        	<arg name="debug" value="$(arg debug)" />
        	<arg name="gui" value="$(arg gui)" />
        	<arg name="paused" value="$(arg paused)"/>
        	<arg name="use_sim_time" value="$(arg use_sim_time)"/>
        	<arg name="headless" value="$(arg headless)"/>
		<!--arg name="world_name" value="$(find racecar_description)/worlds/racetrack.world"/-->
               <!-- .world文件的地址-->
    	</include>

	<!-- 从yaml文件加载联合控制器的参数 -->
	<rosparam file="$(find pkg)/config/ctrl.yaml" command="load"/>

	<!-- 加载机器人模型描述参数 -->
	<group ns="deepracer1">
	<param name="robot_description" textfile="$(find pkg)/urdf/deepracer1.urdf"/>
	<node name= "robot_state_publisher" pkg= "robot_state_publisher" type= "robot_state_publisher">
		<param name="publish_frequency" type="double" value="20.0" />
	</node>
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
          args="-urdf -model deepracer1 -param robot_description -x -1.5 -y -4 -z 0.04"/> 
	<node name="controller_manager" pkg="controller_manager" type="spawner" 
	      respawn="true" output="screen" ns="/deepracer1" 
	      args="left_rear_wheel_velocity_controller       right_rear_wheel_velocity_controller
	            left_front_wheel_velocity_controller      right_front_wheel_velocity_controller
	            left_steering_hinge_position_controller   right_steering_hinge_position_controller
	            joint_state_controller"/>
	</group>

	<!-- <group ns="deepracer2">
	<param name="robot_description" textfile="$(find pkg)/urdf/deepracer2.urdf"/>
	<node name= "robot_state_publisher" pkg= "robot_state_publisher" type= "robot_state_publisher">
		<param name="publish_frequency" type="double" value="20.0" />
	</node>
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
          args="-urdf -model deepracer2 -param robot_description -x 1.5 -y -4 -z 0.04"/> 
	<node name="controller_manager" pkg="controller_manager" type="spawner" 
	      respawn="true" output="screen" ns="/deepracer2" 
	      args="left_rear_wheel_velocity_controller       right_rear_wheel_velocity_controller
	            left_front_wheel_velocity_controller      right_front_wheel_velocity_controller
	            left_steering_hinge_position_controller   right_steering_hinge_position_controller
	            joint_state_controller"/>
	</group>

	<group ns="deepracer3">
	<param name="robot_description" textfile="$(find pkg)/urdf/deepracer3.urdf"/>
	<node name= "robot_state_publisher" pkg= "robot_state_publisher" type= "robot_state_publisher">
		<param name="publish_frequency" type="double" value="20.0" />
	</node>
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
          args="-urdf -model deepracer3 -param robot_description -x 0 -y -4 -z 0.04"/> 
	<node name="controller_manager" pkg="controller_manager" type="spawner" 
	      respawn="true" output="screen" ns="/deepracer3" 
	      args="left_rear_wheel_velocity_controller       right_rear_wheel_velocity_controller
	            left_front_wheel_velocity_controller      right_front_wheel_velocity_controller
	            left_steering_hinge_position_controller   right_steering_hinge_position_controller
	            joint_state_controller"/>
	</group>

	<group ns="deepracer4">
	<param name="robot_description" textfile="$(find pkg)/urdf/deepracer4.urdf"/>
	<node name= "robot_state_publisher" pkg= "robot_state_publisher" type= "robot_state_publisher">
		<param name="publish_frequency" type="double" value="20.0" />
	</node>
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
          args="-urdf -model deepracer4 -param robot_description -x 0 -y -4.5 -z 0.04"/> 
	<node name="controller_manager" pkg="controller_manager" type="spawner" 
	      respawn="true" output="screen" ns="/deepracer4" 
	      args="left_rear_wheel_velocity_controller       right_rear_wheel_velocity_controller
	            left_front_wheel_velocity_controller      right_front_wheel_velocity_controller
	            left_steering_hinge_position_controller   right_steering_hinge_position_controller
	            joint_state_controller"/>
	</group> -->

<!-- <node pkg="pkg" type="servo_commands.py" name="servo_commands" output="screen">
</node> -->

</launch>



